cogs/birthday/birthday.py

```diff
@@ -519,46 +556,119 @@ async def setSelfBirthday(self, ctx: Context, *, birthday: MonthDayConverter):
-                await ctx.author.send(f"{headerBad}: Declined. Not setting your birthday.")
-                return
-
-            await birthdayConfig.get_attr(KEY_BDAY_MONTH).set(birthday.month)
-            await birthdayConfig.get_attr(KEY_BDAY_DAY).set(birthday.day)
-            await birthdayConfig.get_attr(KEY_ADDED_BEFORE).set(True)
+                self.logger.info(
+                    "%s#%s (%s) added their birthday as %s",
+                    ctx.author.name,
+                    ctx.author.discriminator,
+                    ctx.author.id,
+                    birthdayStr,
+                )
 
-            await ctx.author.send(
-                f"{headerGood}: Successfully set your birthday to "
-                f"{spoiler(bold(birthdayStr, escape_formatting=False), escape_formatting=False)}."
-            )
+                # explicitly check to see if user should be added to role, if the month
+                # and day just so happen to be the same as it is now.
+                await self.checkBirthday()
+                return
+            except discord.Forbidden:
 
-            self.logger.info(
-                "%s#%s (%s) added their birthday as %s",
-                ctx.author.name,
-                ctx.author.discriminator,
-                ctx.author.id,
-                birthdayStr,
-            )
+                def check(msg: discord.Message):
+                    return msg.author == ctx.author and msg.channel == ctx.channel
 
-            # explicitly check to see if user should be added to role, if the month
-            # and day just so happen to be the same as it is now.
-            await self.checkBirthday()
-            return
+                try:
+                    await ctx.send(
+                        "You have disabled DMs from this server. Would you "
+                        "still like to continue here?"
+                        f"Type {bold('`yes`', escape_formatting=False)} to confirm."
+                    )
+                    response = await self.bot.wait_for("message", timeout=30.0, check=check)
+                except asyncio.TimeoutError:
+                    await ctx.send(f"{headerBad}: You took too long. Not setting your birthday.")
+                    return
 
+                if response.content.lower() != "yes":
+                    await ctx.send(f"{headerBad}: Declined. Not setting your birthday.")
+                    return
+                can_delete = ctx.channel.permissions_for(ctx.me).manage_messages
+
+                if can_delete:
+                    msg: discord.Message = await ctx.send(f"{headerWarn}: {confirmationStr}")
+                    try:
+                        response = await self.bot.wait_for("message", timeout=10.0, check=check)
+                    except asyncio.TimeoutError:
+                        try:
+                            await ctx.send(f"{headerBad}: No response. Aborting", delete_after=10)
```

Comment:
A nit. Also, I don't think deleting is necessary, but if you think it is better, adjust the other similar one to have `delete_after` as well.
```suggestion
                            await ctx.send(f"{headerBad}: No response. Aborting.")
```

Reply:
Nit has been fixed.
Also I have removed all `delete_after` statements from messages that do not contain sensitive information.